### PR TITLE
PUBDEV-5233: add first batch of Hadoop tests of distributed XGBoost

### PR DESCRIPTION
**Pull request by rpeck**

### CODE REVIEW

File: h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R [hunk 1]
```diff
@@ -0,0 +1,71 @@
+setwd(normalizePath(dirname(R.utils::commandArgs(asValues=TRUE)$"f")))
+source("../../../scripts/h2o-r-test-setup.R")
+
+
+## Data: 5MM records and 500 features
+
+CreateData <- function(nrows, ncols){
+  
+  airlines <- h2o.importFile("hdfs://mr-0xd6.0xdata.loc:8020/datasets/airlines_all.05p.csv")
```
Comment (michalkurka) [inline h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R]: Should not be hardcoded. I think

**Comment by wendycwong** [comment]
One big problem here.  As of today, we do not have Runit tests running on Hadoop off master.  There is a test written for it but it is disabled.

@rpeck:  unless you want to fix this problem, can you rewrite your test in python?  

1. Make sure you put your tests in the following directory: h2o-3/h2o-py/tests/testdir_hdfs.
2. Add your new tests into the test lists in index.list.

Thanks.

**Comment by rpeck** [comment]
@wendycwong: I pushed these so that @michalkurka and @mdymczyk could more easily run them manually.  MK is working on getting units to run under Hadoop in Jenkins.  Once that's ready I'm going to copy these to testdir_hdfs so that they run under both raw h2o clusters and Hadoop, in Jenkins.

In the next day or two I'm going to push some Python tests as well.

File: h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R (hunk 1 above)
Comment (wendycwong) [reply h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R]: We have about 3 hadoop clusters with different name nodes.  Tests will fail if you hard code the path to your file.

**Comment by wendycwong** [comment]
Ray:

That will be great.  The python ones should be easier as the jenkins are already set up for it.


Wendy

File: h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R (hunk 1 above)
Comment (rpeck) [reply h2o-r/tests/testdir_algos/xgboost/runit_INTERNAL_XGBoostEstimation.R]: Ok. . .  I took this right from Megan's original code.  So, what should I change this to?

**Pull request merged**